File: UVa problems/11526 H(n).cpp
//
// Created by maryam on 8/28/17.
//
//#define Q11526
#ifdef Q11526

#include "bits/stdc++.h"
using namespace std;
typedef long long int ll;

int main(){
    ios::sync_with_stdio(0);
    int t;cin >> t;
    while(t--) {
        ll n;
        cin >> n;
        if(!n) cout << 0 << endl;
        else if(n == 1) cout << 1 << endl;
        else if(n == 2) cout << 3 << endl;
        else if(n == 3) cout << 5 << endl;
        else {
            ll sum = n;
            ll num = n, pre = 0;
            ll i = 2, mul = 0;
            ll till = (ll) sqrt(n) - 1;
            for (int tt = 0; tt < till; tt++) {
                pre = n / (i++);
                num -= pre;
                sum += (num * (++mul));
                sum += pre;
                num = pre;
            }
            sum += ((mul + 1) * (pre - till - 1));
            cout << sum << endl;
        }
    }

    return 0;
}

#endif

File: UVa problems/11526H(n).py
import math
t = int(input())

for i in range(t):
    n = int(input())
    if not n: print(0)
    elif n == 1: print(1)
    elif n == 2: print(3)
    elif n == 3: print(5)
    else:
        sum = num = n
        pre = 0
        i = 2
        mul = 0
        till = int(math.sqrt(n))
        #print(till)
        for tt in range(till):
            pre = n // (i)
            i = i + 1
            num = num - pre
            sum = sum + (num * (++mul))
            sum = sum + pre
            num = pre

        sum = sum + ((mul + 1) * (pre - till - 1))
        print(int(sum))



